plugins/plextor/qscan_cmd_plextor.cpp

```diff
@@ -0,0 +1,795 @@
+  } else {
+    data->jitter = 3600 - (int)(2.4 * (data->jitter));
+  }
+  return 0;
+  //	return (!(dev->rd_buf[2]));
+}
+
+int scan_plextor::cmd_fete_getdata() {
+  dev->cmd[0] = PLEXTOR_FETE_READOUT; // 0xF5;
+  dev->cmd[3] = 0x0C;
+  dev->cmd[9] = 0xCE;
+  if ((dev->err = dev->cmd.transport(READ, dev->rd_buf, 0xCE))) {
+    sperror("PLEXTOR_FETE_READOUT", dev->err);
+    return dev->err;
+  }
+#ifdef _debug_fete
+  for (int i = 0; i < 0xCE; i++) {
+    if (!(i % 0x20))
+      printf("\n");
+    printf("%02X ", dev->rd_buf[i] & 0xFF);
+  }
+  printf("\n");
+#endif
+  return 0;
+}
+
+int scan_plextor::cmd_fete_block(cdvd_ft *data) {
+  int rdy = test_unit_ready(dev);
+  int offs = ((++fete_idx) << 1) + 8;
+  if (rdy != 0x20408) {
+    printf("test unit ready = %05X, return\n", rdy);
+    return -1;
+  }
+  //	if (cmd_fete_getdata()) return -1;
+  //	data->te=dev->rd_buf[offs];
+  //	data->fe=dev->rd_buf[offs+1];
+  data->te = 0;
+  data->fe = 0;
+  if (fete_idx > 99)
+    return 1;
+  lba = (fete_idx + 1) * fete_rsize;
+  if (lba > (dev->media.capacity_total - 1))
+    lba = dev->media.capacity_total - 1;
+
+  while ((data->fe == 0 || data->te == 0) && (fete_idx <= 99) &&
+         (rdy == 0x20408)) {
+    if (cmd_fete_getdata())
+      return -1;
+    data->te = dev->rd_buf[offs];
+    data->fe = dev->rd_buf[offs + 1];
+    if (data->fe == 0 || data->te == 0) {
+      rdy = test_unit_ready(dev);
+#ifdef _debug_fete
+      printf("test unit ready = %05X\n", rdy);
+#endif
+      msleep(10);
+    }
+  }
+  return 0;
+}
+
+int scan_plextor::build_TA_histogram_px716(unsigned char *response_data,
+                                           int *dest_pit, int *dest_land,
+                                           int len) {
+  int *dest[] = {dest_land, dest_pit};
+  int count = ntoh16(response_data + 2);
+  //	printf("PX-716 Histogram... %d\n",count);
+  int idx = 28;
+  int v, pit;
+  for (int i = 0; i < count; i++) {
+    v = ntoh16u(response_data + idx);
+    pit = !!(v & 0x8000);
+    v &= ~0x8000;
+    dest[pit][min(v, len - 1)]++;
+    idx += 2;
+  }
+  return 0;
+}
+
+int scan_plextor::build_TA_histogram_px755(unsigned char *response_data,
+                                           int *dest_pit, int *dest_land,
+                                           int len, int dt) { //, int spd) {
+  int *dest[] = {dest_land, dest_pit};
+  int count = ntoh16(response_data + 2);
+  //	printf("PX-755/760 Histogram... %d\n", count);
+  int idx = 28;
+  int v, pit;
+  for (int i = 0; i < count; i++) {
+    v = ntoh16u(response_data + idx);
+    pit = !!(v & 0x8000);
+    v &= ~0x8000;
+    if (dt & DISC_DVDplus)
+      dest[pit][min((int)(v * 1.452), len - 1)]++; // DVD+R(W), speed 2.4x
+    else
+      dest[pit][min((int)(v * 1.21), len - 1)]++; // DVD-R(W), speed 2x
+    idx += 2;
+  }
+  return 0;
+}
+
+int scan_plextor::evaluate_histogramme(cdvd_ta *data, int **peaks, int **mins) {
+  int i, j1, j2;
+  int local_max = 0;
```

Comment:
Codacy found an issue: [The scope of the variable 'local_max' can be reduced.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6072101)